docs/quickstarts/nextjs.mdx

```diff
@@ -277,83 +268,113 @@ export default function Example() {
 
-In some cases, you may need the full [`User`](/docs/references/javascript/user/user) object. For example, if you want to access the user's email address or name. You can use the [`clerkClient`](/docs/references/backend/overview) helper to get the full `User` object.
+<Callout type="info">
+  Any requests from a Client Component to a Route Handler will automatically include the user's current token as the Bearer token.
+</Callout>
 
+<Tabs items={["Route Handler", "Route Handler w/ User Fetch"]}>
+  <Tab>
+    You can protect your API routes by using the [`auth()`](/docs/references/nextjs/auth) helper, if the route is not protected by [middleware](/docs/references/nextjs/auth-middleware). 
 
-```tsx filename="pages/api/auth.ts"
-import { clerkClient } from "@clerk/nextjs";
-import { getAuth } from "@clerk/nextjs/server";
-import type { NextApiRequest, NextApiResponse } from "next";
+    ```tsx filename="app/api/user/route.[jsx/tsx]"
+    import { NextResponse } from 'next/server';
+    import { auth } from '@clerk/nextjs';
 
-export default async function handler(
-  req: NextApiRequest,
-  res: NextApiResponse
-) {
-  const { userId } = getAuth(req);
+    export async function GET(req: Request, res: NextResponse) {
+      const { userId } = auth();
 
-  if (!userId) {
-    return res.status(401).json({ error: "Unauthorized" });
-  }
+      if (!userId) {
+        return new NextResponse("Unauthorized", { status: 401 });
+      }
+
+      // Perform your Route Handler's logic
 
-  const user = userId ? await clerkClient.users.getUser(userId) : null;
+      return NextResponse.json({ userId }, { status: 200 });
+    }
+    ```
+  </Tab>
+  <Tab>
+    You can protect your API routes by using the [`auth()`](/docs/references/nextjs/auth) helper, if the route is not protected by [middleware](/docs/references/nextjs/auth-middleware). 
 
-  // use the user object to decide what data to return
+    In some cases, you may need the full [`User`][user-object] object. For example, if you want to access the user's email address address or name, you can use  [`currentUser()`](/docs/references/nextjs/current-user) helper to get the full [`User`][user-object] object. Under the hood, this calls `fetch()` so it is automatically deduped per request.
 
-  return res.status(200).json({});
-}
-```
+   ```tsx filename="app/api/user/route.ts"
+    import { NextResponse } from 'next/server';
+    import { currentUser, auth } from "@clerk/nextjs";
 
-### Server side render user data
+    export async function GET(req: Request, res: NextResponse) {
+      const { userId } = auth();
 
-<Tabs items={["App Router", "Pages Router"]}>
+      if (!userId) {
+        return new NextResponse("Unauthorized", { status: 401 });
+      }
+
+      const user = await currentUser();

```

Comment:
should we being doing something like

```
const user = currentUser();

if (!user) { return new NextResponse("Unauthorized", { status: 401 }) }

// Perform Route Handler logic with the returned user object
```

Reply:
I was mulling that over, versus keeping some use of auth() in the example. We have the other tab though so this is fine.